<a href="https://colab.research.google.com/github/JohnLloyd31/CPE311/blob/main/Seatwork_6_1_Exploratory_Data_Analysis_on_Your_Own_Dataset.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Seatwork 6.1 Exploratory Data Analysis on Your Own Dataset


Instructions:

- Choose your own dataset from public repositories.
- Link your chosen dataset in this spreadsheet: Dataset SpreadsheetLinks to an  external site.
- Perform simple exploratory data analysis using descriptive statistics.
- Employ the same steps as done in the hands-on activity and provide your own - analysis of the dataset.
- Your submission must include your GitHub folder link that includes a customized readme file, Python Notebook Files, Dataset, and a simple presentation of your findings.

In [2]:
import pandas as pd
import numpy as np
filepath = '/content/Student Mental health.csv'
mentalHealth_df=pd.read_csv(filepath)
mentalHealth_df


Timestamp Choose your gender   Age What is your course?  \
0         8/7/2020 12:02             Female  18.0          Engineering   
1         8/7/2020 12:04               Male  21.0    Islamic education   
2         8/7/2020 12:05               Male  19.0                  BIT   
3         8/7/2020 12:06             Female  22.0                 Laws   
4         8/7/2020 12:13               Male  23.0         Mathemathics   
..                   ...                ...   ...                  ...   
96   13/07/2020 19:56:49             Female  21.0                  BCS   
97   13/07/2020 21:21:42               Male  18.0          Engineering   
98   13/07/2020 21:22:56             Female  19.0             Nursing    
99   13/07/2020 21:23:57             Female  23.0     Pendidikan Islam   
100  18/07/2020 20:16:21               Male  20.0   Biomedical science   

    Your current year of Study What is your CGPA? Marital status  \
0                       year 1        3.00 - 3.49             No   
1                       year 2        3.00 - 3.49             No   
2                       Year 1        3.00 - 3.49             No   
3                       year 3        3.00 - 3.49            Yes   
4                       year 4        3.00 - 3.49             No   
..                         ...                ...            ...   
96                      year 1        3.50 - 4.00             No   
97                      Year 2        3.00 - 3.49             No   
98                      Year 3        3.50 - 4.00            Yes   
99                      year 4        3.50 - 4.00             No   
100                     Year 2        3.00 - 3.49             No   

    Do you have Depression? Do you have Anxiety? Do you have Panic attack?  \
0                       Yes                   No                       Yes   
1                        No                  Yes                        No   
2                       Yes                  Yes                       Yes   
3                       Yes                   No                        No   
4                        No                   No                        No   
..                      ...                  ...                       ...   
96                       No                  Yes                        No   
97                      Yes                  Yes                        No   
98                      Yes                   No                       Yes   
99                       No                   No                        No   
100                      No                   No                        No   

    Did you seek any specialist for a treatment?  
0                                             No  
1                                             No  
2                                             No  
3                                             No  
4                                             No  
..                                           ...  
96                                            No  
97                                            No  
98                                            No  
99                                            No  
100                                           No  

[101 rows x 11 columns]

In [3]:
print(mentalHealth_df.columns)

Index(['Timestamp', 'Choose your gender', 'Age', 'What is your course?',
       'Your current year of Study', 'What is your CGPA?', 'Marital status',
       'Do you have Depression?', 'Do you have Anxiety?',
       'Do you have Panic attack?',
       'Did you seek any specialist for a treatment?'],
      dtype='object')


In [4]:
print(mentalHealth_df.dtypes)

Timestamp                                        object
Choose your gender                               object
Age                                             float64
What is your course?                             object
Your current year of Study                       object
What is your CGPA?                               object
Marital status                                   object
Do you have Depression?                          object
Do you have Anxiety?                             object
Do you have Panic attack?                        object
Did you seek any specialist for a treatment?     object
dtype: object


In [48]:
t_records = mentalHealth_df.shape[0]
print("Total number of records: ", t_records)

Total number of records:  101


In [65]:
Ydepression = mentalHealth_df[mentalHealth_df['Do you have Depression?'] == 'Yes']
Ydepression.shape[0]

35

In [50]:
percentage_depression = (Ydepression.shape[0] / t_records) *100
print("Percentage of student that have depression:",round(percentage_depression, 2),"%")

Percentage of student that have depression: 34.65 %


In [51]:
Ndepression = mentalHealth_df[mentalHealth_df['Do you have Depression?'] == 'No']
Ndepression.shape[0]

66

In [52]:
Npercentage_depression = (Ndepression.shape[0] / t_records) * 100
print("Percentage of student that don't have depression:",round(Ndepression.shape[0], 2),"%")

Percentage of student that don't have depression: 66 %


In [53]:
Yanxiety = mentalHealth_df[mentalHealth_df['Do you have Anxiety?'] == 'Yes']
Yanxiety.shape[0]

34

In [54]:
percentage_anxiety = (Yanxiety.shape[0] / t_records) *100
print("Percentage of student that have axiety:",round(percentage_anxiety, 2),"%")

Percentage of student that have axiety: 33.66 %


In [55]:
Nanxiety = mentalHealth_df[mentalHealth_df['Do you have Anxiety?'] == 'No']
Nanxiety.shape[0]

67

In [56]:
Npercentage_anxiety = (Nanxiety.shape[0] / t_records) *100
print("Percentage of student that have don't axiety:",round(Npercentage_anxiety, 2),"%")

Percentage of student that have don't axiety: 66.34 %


In [57]:
Ypanic = mentalHealth_df[mentalHealth_df['Do you have Panic attack?'] == 'Yes']
Ypanic.shape[0]

33

In [58]:
percentage_panic = (Ypanic.shape[0] / t_records) *100
print("Percentage of student that have have panic attack:",round(percentage_panic, 2),"%")

Percentage of student that have have panic attack: 32.67 %


In [59]:
Npanic = mentalHealth_df[mentalHealth_df['Do you have Panic attack?'] == 'No']
Npanic.shape[0]

68

In [60]:
Npercentage_panic = (Npanic.shape[0] / t_records) *100
print("Percentage of student that have have don't panic attack:",round(Npercentage_panic, 2),"%")

Percentage of student that have have don't panic attack: 67.33 %


In [61]:
Ytreatment = mentalHealth_df[mentalHealth_df['Did you seek any specialist for a treatment?'] == 'Yes']
Ytreatment.shape[0]

6

In [62]:
percentage_treatment = (Ytreatment.shape[0] / t_records) * 100
print("Percentage of student that seek specialist for a treatment:",round(percentage_treatment, 2),"%")

Percentage of student that seek specialist for a treatment: 5.94 %


In [63]:
Ntreatment = mentalHealth_df[mentalHealth_df['Did you seek any specialist for a treatment?'] == 'No']
Ntreatment.shape[0]

95

In [64]:
Npercentage_treatment = (Ntreatment.shape[0] / t_records) * 100
print("Percentage of student that seek specialist for a treatment:",round(Npercentage_treatment, 2),"%")

Percentage of student that seek specialist for a treatment: 94.06 %


#Analysis
- According to the statistics, a small percentage of students suffer from depression, anxiety, or panic attacks, accounting for around 34.65%, 33.66%, and 32.67%, respectively. However, just 5.94% of students seek specialized help for these problems. This is a large disparity between the incidence of mental health issues and the use of professional care among students. Efforts to raise mental health awareness and provide access to support services on campuses are critical to resolving this imbalance and effectively supporting student well-being.